src/server/server/dal.py

```diff
@@ -133,61 +133,29 @@ def _get_multi(self, keys):
         key_to_entity = dict((entity.key, entity) for entity in entities)
         return [key_to_entity[key] for key in keys]
 
-    def get_user(self, email):
-        """
-        :param email:
-        :return: Struct.User if user found, else None
-        """
-        (user, _) = self.get_user_and_password(email)
-        return user
 
-    def get_user_and_password(self, email):
-        """
-        :param email:
-        :return: A tuple (Struct.User, encoded password) if user found, else (None, None)
-        """
-        key = self._ds_client.key('User', email)
-        db_user = self._ds_client.get(key)
-        return (_to_user(db_user), _to_user_password(db_user)) if db_user else (None, None)
+class Dal(object):
+    # Dal is a class rather than plain module functions to enable init logic in the future,
+    # but datastore client is slow to initialize, so we share it between Dal instances
 
-    def get_all_users(self):
-        """
-        :return: A list of struct.User of all users in database
-        """
-        all_users_query = self._ds_client.query(kind='User').fetch()
-        return [_to_user(db_user) for db_user in all_users_query]
+    _ds_client = _datastore_client()
 
-    def save_user(self, user, password):
-        """
-        save a user into the database, if it's newly created, db_keys will be initialized
-        """
-        if user._user_doc_set_db_key is None:  # pylint: disable=protected-access
-            user_doc_set_db = self._make_entity('UserDocumentSet', not_indexed=('user_documents',))
-            user_doc_set_db['user_documents'] = []
-            self._ds_client.put(user_doc_set_db)
-            user._user_doc_set_db_key = user_doc_set_db.key  # pylint: disable=protected-access
-        if user._user_computed_profile_db_key is None:  # pylint: disable=protected-access
+    def __init__(self):
+        ds_helper = DatastoreHelper(self._ds_client)
+        self.feature_set = DalFeatureSet(self._ds_client, ds_helper)
```

Comment:
_ds_client is passed twice, as a parameter and in ds_helper, ds_helper._ds_client could be used